### ISSUE TITLE
App shows blank screen on Arch

### ISSUE THREAD

**Issue by anpang54** [modrinth/code#3062]
### Please confirm the following.

- [x] I checked the [existing issues](https://github.com/modrinth/code/issues) for duplicate problems
- [x] I have tried resolving the issue using the [support portal](https://support.modrinth.com)
- [x] I have ensured my Modrinth App installation is up to date

### What version of the Modrinth App are you using?

0.9.0

### What operating systems are you seeing the problem on?

Linux

### Describe the bug

when I open the modrinth app AppImage, it loads for a while and then just shows a blank/white screen

when I run the app from the terminal, I get:
```
Could not create default EGL display: EGL_BAD_PARAMETER. Aborting...

** (ModrinthApp:41530): WARNING **: 19:04:47.326: atk-bridge: get_device_events_reply: unknown signature
```

this is the same error as #3057, but it's not an nvidia gpu and instead of being laggy, it shows a blank screen and no ui elements at all

the OS is arch linux with wayland and kde plasma, with intel integrated graphics
I've tried `WEBKIT_DISABLE_DMABUF_RENDERER=1`, changing to x11, and updating the entire system and the modrinth app, but nothing works

### Steps to reproduce

1. be on arch linux with wayland, kde, intel graphics
2. open modrinth
3. the app just shows a blank screen

### Expected behavior

the app should launch normally like it did

**Comment by anpang54** [comment]
also happens in fedora

**Comment by lxmadabc** [comment]
Same here behaviour on manjaro (arch) with latest updates and AMD graphics. Latest update was installed on wayland, also tested x11 as some suggested in other repos that this might be the issue.

**Comment by glitchy-sk8k9** [comment]
Can reproduce on hyprland intel GPU. Weirdly 8.x won't work now either. I'm on appimage.

**Comment by glitchy-sk8k9** [comment]
```➜  ~ cat /home/glitchy/.local/share/ModrinthApp/launcher_logs/session_20241224_184615.log
2024-12-24T18:46:15.974158381+00:00  INFO theseus_gui: Initialized tracing subscriber. Loading Modrinth App!
2024-12-24T18:46:16.000543988+00:00  INFO theseus_gui: Initializing app...```

Anyone else have just these logs?

**Comment by glitchy-sk8k9** [comment]
Not issue with a corrupted `~/.local/share/ModrinthApp`, i mv'd everything to give a fresh start, same error.

**Comment by lxmadabc** [comment]
> ```
> 2024-12-24T18:46:15.974158381+00:00  INFO theseus_gui: Initialized tracing subscriber. Loading Modrinth App!
> 2024-12-24T18:46:16.000543988+00:00  INFO theseus_gui: Initializing app...
> ```
> 
> Anyone else have just these logs?

Yes, mine are the same. Also tried removing the local data and downloaded the most recent appimage with no luck.

**Comment by cranberry3148** [comment]
Arch users can try the `modrinth-app-segfault-fix-bin` AUR package for a quick fix, I've been using it myself ever since 0.8.0. Users on other distros can try setting these env variables:
```bash
export WEBKIT_DISABLE_DMABUF_RENDERER=1
export LIBGL_ALWAYS_SOFTWARE=1
export GDK_BACKEND=x11
```

**Comment by sehrschlechtYT** [comment]
Same error also occurs on Fedora 41 with Wayland and KDE Plasma.

**Comment by glitchy-sk8k9** [comment]
@cranberry3148 
> Arch users can try the `modrinth-app-segfault-fix-bin` AUR package for a quick fix, I've been using it myself ever since 0.8.0. 

This fixed the issue for me.

**Comment by glitchy-sk8k9** [comment]
https://github.com/modrinth/code/issues/3057#issuecomment-2561077541 fixes lag

TLDR: 
1. Set WEBKIT_DISABLE_DMABUF_RENDERER to 0
2. Run /opt/modrinth-app/modrinth-app
3. Profit.

Why does this work? I have no sweet clue. @cranberry3148 those env variables were already overwritten by the shell script.

```
➜  ~ cat /bin/modrinth-app 
#!/bin/sh

export WEBKIT_DISABLE_DMABUF_RENDERER=1
export LIBGL_ALWAYS_SOFTWARE=1
export GDK_BACKEND=x11

# Launch the modrinth app
exec /opt/modrinth-app/modrinth-app "$@"
```

**Comment by anpang54** [comment]
the `modrinth-app-segfault-fix-bin` package doesn't work for me, when I opened the app it showed this error:
> Error initializing Modrinth App
> Modrinth App failed to load correctly. This may be because of a corrupted file, or because the app is missing crucial files.

setting the 3 environment variables also didn't work for me, it just showed a blank screen like before

**Comment by anpang54** [comment]
for the aur package, when I run the executable in a command line it throws this error
```
Permission denied (os error 13) at path "/opt/modrinth-app/tauri_current/tauri_current_appY9k605"
```
it's not some permission error because the path `/opt/modrinth-app/tauri_current` doesn't even exist

**Comment by glitchy-sk8k9** [comment]
new update f*cked something ome too ![Image](https://github.com/user-attachments/assets/347e16d8-9208-4569-9f0e-0307988bd58b)

**Comment by anpang54** [comment]
yep

**Comment by LOENS2** [comment]
Same here for me.

**Comment by LOENS2** [comment]
Unfortunately, due to database migrations it's not possible to use an older version of the app which last worked for me.
![Image](https://github.com/user-attachments/assets/0056dcf7-438e-4cc5-8350-d5cb9729a669)

**Comment by LOENS2** [comment]
I found a way to be able to play with the old version with the help of a backup I created before updating Modrinth.

1. Restore the backup of `~/.local/share/Modrinth-App`
2. Download the last working version of Modrinth (for me [0.8.9](https://launcher-files.modrinth.com/versions/0.8.9/linux/Modrinth%20App_0.8.9_amd64.AppImage))
3. Turn of the internet connection
4. Start the Modrinth appimage
5. Export the modpack to a .mrpack file
6. Close the app and delete the folder `~/.local/share/Modrinth-App`
7. Reopen the app
8. Turn on the internet
9. Import the .mrpack

I know that this isn't practical, but it works pretty well for a workaround right now.

**Comment by glitchy-sk8k9** [comment]
`Permission denied (os error 13) at path "/opt/modrinth-app/tauri_current/tauri_current_appY9k605"`

Anyone with a brain can use this info?

**Comment by Mathew321** [comment]
This is what I did to solve the same issue.

Step 1:
```wget https://archive.archlinux.org/packages/m/mesa/mesa-1%3A24.0.5-1-x86_64.pkg.tar.zst```

Step 2:
```sudo pacman -U ./mesa-1:24.0.5-1-x86_64.pkg.tar.zst```

Step 3:
```sudo nano /etc/pacman.conf```
or
```sudo vi /etc/pacman.conf```
To save in vi press CTRL + C together and write :wq!

Step 4:
Add to the ignored packages mesa in the /etc/pacman.conf:
```IgnorePkg = mesa```

Step 5:
```yay -S llvm17```

Try to install this package and if it doesn't work you can try running
```sudo pacman -Syy```
first.

Step 6:
Try running the AppImage afterwords!
This worked for me doesn't mean it will work as well for you.

NOTE: I came across some weird errors while running it in terminal.
Something about 
```~/.drirc having an error in line -1 column -1: No element found```
I'm sorry but that error is way beyond my paygrade and I have no idea what that error is! 

I did almost the same thing on fedora.
Step 1:
```sudo dnf5 downgrade mesa-libEGL```
Step 2:
There is no step two it's just works after lol :)

**Comment by anpang54** [comment]
when I downgraded mesa, the next time I started the computer the sddm-greeter broke
seems like the new mesa is incompatible with modrinth and the old mesa is incompatible with sddm-greeter

**Comment by LOENS2** [comment]
It seems to work fine again with the latest update of the AUR package "modrinth-app-bin"

**Comment by anpang54** [comment]
the problem is randomly back now, even with the arch package

**Comment by i-am-called-glitchy** [comment]
> the problem is randomly back now, even with the arch package

Vouch.

**Comment by dylanparry** [comment]
Edit the file file at `/usr/bin/modrinth-app` (eg. `sudo nano /usr/bin/modrinth-app`) and replace the value of `WEBKIT_DISABLE_DMABUF_RENDERER` with `0` (it should be `1` to begin with).

That will solve the problem now, but it'll still happen again in the future when there's another update... so you can also run `sudo chattr +i /usr/bin/modrinth-app` to mark the file as immutable, so even root won't overwrite it. That should stop updates from changing the file, which is after all only a shell script pointing to the real executable and doesn't really do anything special.

**Comment by anpang54** [comment]
when I do that, the app still shows the blank screen with the same
```
Could not create default EGL display: EGL_BAD_PARAMETER. Aborting...
```

**Comment by cranberry3148** [comment]
I don't expect them to do anything about this as I've read somewhere on Discord that they're not interested in supporting Linux themselves, so I've switched back to ATLauncher. I am not sure if that info is outdated.

**Comment by triphora** [comment]
This is not true in the slightest. We are still supporting Linux. However, it can take us longer to fix issues for Linux since none of our app developers daily drive the system. As a result, some of the issues have fallen to the wayside a bit. As a Linux user myself (Arch btw) I will be working with the developers to ensure we improve the experience for Linux users going forward.

**Comment by i-am-called-glitchy** [comment]
> This is not true in the slightest. We are still supporting Linux. However, it can take us longer to fix issues for Linux since none of our app developers daily drive the system. As a result, some of the issues have fallen to the wayside a bit. As a Linux user myself (Arch btw) I will be working with the developers to ensure we improve the experience for Linux users going forward.

incoming PR BS

**Comment by Mathew321** [comment]
Guys, I had the same issues like you, but I just fixed every single one of them within like 10-15 minutes. I call this skill issue honestly. Modrinth is free and an open-source community project, so no one is forced to fix your issues on linux in my opinion. If you can't fix this yourself then why are you on Arch in the first place lol.

**Comment by blryface** [comment]
lmao???

oh my god what is this thread

**Comment by i-am-called-glitchy** [comment]
> Guys, I had the same issues like you, but I just fixed every single one of them within like 10-15 minutes. I call this skill issue honestly. Modrinth is free and an open-source community project, so no one is forced to fix your issues on linux in my opinion. If you can't fix this yourself then why are you on Arch in the first place lol.

you proud of yourself? <3

**Comment by blryface** [comment]
inb4 locked for too heated

**Comment by i-am-called-glitchy** [comment]
aight

**Comment by Arch-User-BTW** [comment]
I created an account just because you guys were so obnoxious (I use arch btw). Y'all are so immature, this sounds like a 3rd grade class.

**Comment by i-am-called-glitchy** [comment]
> I created an account just because you guys were so obnoxious (I use arch btw). Y'all are so immature, this sounds like a 3rd grade class.

my guy, shut up + ratio
